tinygrad/runtime/ops_hip.py

```diff
@@ -106,6 +99,18 @@ def __init__(self, name:str, prg:str, binary=False):
       hip.hipSetDevice(i)
       self.prgs.append(hip.hipModuleGetFunction(hip.hipModuleLoadData(prg), name))
 
+  @cache_compiled
+  def compile(self, prg, name):
+    output_file = pathlib.Path(tempfile.mktemp())
+    try:
+      prog = hip.hiprtcCreateProgram(prg, name, [], [])
+      hip.hiprtcCompileProgram(prog, [f'--offload-arch={hip.hipGetDeviceProperties(HIP.default_device).gcnArchName}'])
+      output_file.write_bytes(hip.hiprtcGetCode(prog))
```

Comment:
Hmm, now this writes a file if I don't cache.

Like I'm not being pedantic here, this introduces complexity that doesn't make sense. While it's readable, why?